index.ts

```diff
@@ -0,0 +1,303 @@
+      const myHeaders = new Headers();
+      myHeaders.append(
+        "Authorization",
+        "Bearer ",
+      ),
+      myHeaders.append("Content-Type", "application/json");
+      const raw = JSON.stringify({
+        model: "gpt-3.5-turbo",
+        messages: [
+                {
+                  role: "user",
+                  content: prompt,
+                },
+              ],
+        temperature: 0,
+        max_tokens: max_tokens,
+      });
+    
+      const response = fetch("https://api.openai.com/v1/chat/completions", {
+        method: "POST",
+        headers: myHeaders,
+        body: raw,
+        redirect: "follow",
+      })
+        .then((response) => response.json())
+        .then((response) => response.choices[0].message.content);
+    
+      return response;
+    }
+
+    async function fetchFromChatGPT(prompt: string, max_tokens: number) {
+      const myHeaders = new Headers();
+      myHeaders.append(
+        "Authorization",
+        "Bearer ",
+      ),
+      myHeaders.append("Content-Type", "application/json");
+      const raw = JSON.stringify({
+        model: "gpt-3.5-turbo",
+        messages: [
+                {
+                  role: "user",
+                  content: prompt,
+                },
+              ],
+        temperature: 0,
+        max_tokens: max_tokens,
+      });
+    
+      const response = fetch("https://api.openai.com/v1/chat/completions", {
+        method: "POST",
+        headers: myHeaders,
+        body: raw,
+        redirect: "follow",
+      })
+        .then((response) => response.json())
+        .then((response) => response.choices[0].message.content);
+    
+      return response;
+    }
+    const ToLowerCase = (object: string) => {
+      return object.toLowerCase().replace(/\.$/g, "");
+        };
+    const lowerCaseGrammar = Object.keys(grammar).reduce((acc, key) => {
+      acc[ToLowerCase(key)] = grammar[key];
+      return acc;
+    }, {});
+
+// machine
+// ... (previous code)
+
+// machine
+const dmMachine = createMachine(
+  
+  {
+    id: "root",
+    type: "parallel",
+    states: {
+      DialogueManager: {
+        initial: "Prepare",
+        states: {
+          Prepare: {
+            on: { ASRTTS_READY: "Start" },
+            entry: [
+              assign({
+                spstRef: ({ spawn }) => {
+                  return spawn(speechstate, {
+                    input: {
+                      settings: settings,
+                    },
+                  });
+                },
+              }),
+            ],
+          },
+          Start: {
+            initial: "Greeting",
+            states: {
+              Greeting: {
+                entry: "speak.greeting",
+                on: { SPEAK_COMPLETE: "music" },
+              },
+              music: {
+                entry: "speak.music",
+                on: { SPEAK_COMPLETE: "Ask" },
+              },
+              Ask: {
+                entry: "listen",
```

Comment:
```suggestion
                entry: listen(),
```
because you are referring to the function, not pre-defined action. You can predefine it inside actions, if you wish.